pkg/controller/operatingsystemconfig/actuator.go

```diff
@@ -80,14 +79,19 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, osc *extensio
 
 	extensionFiles := getExtensionFiles(osc, networkIsolation)
 
-	osc.Spec.Files = EnsureFiles(osc.Spec.Files, extensionFiles...)
+	switch purpose := osc.Spec.Purpose; purpose {
+	case extensionsv1alpha1.OperatingSystemConfigPurposeProvision:
+		osc := osc.DeepCopy()
+		osc.Spec.Files = EnsureFiles(osc.Spec.Files, extensionFiles...)
 
-	cloudConfig, command, err := oscommonactuator.CloudConfigFromOperatingSystemConfig(ctx, log, a.client, osc, generator.IgnitionGenerator())
-	if err != nil {
-		return nil, nil, nil, nil, nil, nil, fmt.Errorf("could not generate cloud config: %w", err)
-	}
+		userData, err := ignition.New(log).Transpile(osc)
+		return userData, nil, nil, err
 
-	return cloudConfig, command, oscommonactuator.OperatingSystemConfigUnitNames(osc), oscommonactuator.OperatingSystemConfigFilePaths(osc), nil, extensionFiles, nil
+	case extensionsv1alpha1.OperatingSystemConfigPurposeReconcile:
+		return nil, nil, extensionFiles, nil
```

Comment:
is this always sufficient? Even if we would add new extension files as part of an upgrade of the os-metal-extension?
Maybe we should use `EnsureFiles` here, too?